packages/block-editor/src/components/link-control/index.js

```diff
@@ -313,27 +416,49 @@ function LinkControl( {
 			ref={ wrapperNode }
 			className="block-editor-link-control"
 		>
-			{ isEditingLink || ! value ? (
+			{ isResolvingLink && (
+				<div className="block-editor-link-control__loading">
+					{ __( 'Loading' ) }...
+				</div>
+			) }
+
+			{ ( isEditingLink || ! value ) && ! isResolvingLink && (
 				<LinkControlSearchInput
 					value={ inputValue }
 					onChange={ onInputChange }
-					onSelect={ ( suggestion ) => {
-						onChange( { ...value, ...suggestion } );
+					onSelect={ async ( suggestion ) => {
+						if (
+							suggestion.type &&
+							CREATE_TYPE === suggestion.type
+						) {
+							// Await the promise to ensure `stopEditing` is not called prematurely.
+							await handleOnCreate( inputValue );
+						} else {
+							handleSelectSuggestion( suggestion, value )();
+						}
+
+						// Must be called after handling to ensure focus is
+						// managed correctly.
 						stopEditing();
```

Comment:
Are we guaranteed that the component is still mounted by this point? Seems we run the risk of calling `setState` on an unmounted component.

Related:

- https://medium.com/trabe/avoid-updates-on-unmounted-react-components-2fbadab17ad2
- https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html

Reply:
I've looked to address this by wrapping the two pending promises to make them cancellable. Then I'm cancelling them in a `useEffect()`. Then in the `.catch()` of each Promise I'm checking if they were cancelled and if so then I'm bailing out of state updates.